Lifespan tests are broken

### Checklist

<!-- Please make sure you check all these items before submitting your bug report. -->

- [x] The bug is reproducible against the latest release and/or `master`.
- [x] There are no similar issues or pull requests to fix it yet.

### Describe the bug

<!-- A clear and concise description of what the bug is. -->
Some lifespan tests are passing fine while they should fail

### To reproduce

this one for instance pass while the log suggests it should fail, if you change the assert scope to assert False it still passes !

```
@pytest.mark.parametrize("mode", ("auto", "on"))
def test_lifespan_scope_asgi3app(mode):
    async def asgi3app(scope, receive, send):
        assert scope == {"version": "3.0", "spec_version": "2.0"}

    async def test():
        config = Config(app=asgi3app, lifespan=mode)
        lifespan = LifespanOn(config)

        await lifespan.startup()
        await lifespan.shutdown()

    loop = asyncio.new_event_loop()
    loop.run_until_complete(test())
```

```
/home/lotso/PycharmProjects/uvicorn/venv/bin/python /home/lotso/.local/share/JetBrains/Toolbox/apps/PyCharm-P/ch-0/203.5981.165/plugins/python/helpers/pycharm/_jb_pytest_runner.py --target test_lifespan.py::test_lifespan_scope_asgi3app
Testing started at 2:47 PM ...
Launching pytest with arguments test_lifespan.py::test_lifespan_scope_asgi3app in /home/lotso/PycharmProjects/uvicorn/tests

============================= test session starts ==============================
platform linux -- Python 3.8.5, pytest-6.2.0, py-1.10.0, pluggy-0.13.1 -- /home/lotso/PycharmProjects/uvicorn/venv/bin/python
cachedir: .pytest_cache
rootdir: /home/lotso/PycharmProjects/uvicorn, configfile: setup.cfg
plugins: reraise-1.0.3, asyncio-0.14.0
collecting ... collected 2 items

test_lifespan.py::test_lifespan_scope_asgi3app[auto] 
test_lifespan.py::test_lifespan_scope_asgi3app[on] 

============================== 2 passed in 0.11s ===============================

Process finished with exit code 0
PASSED              [ 50%]INFO:     Waiting for application startup.
INFO:     ASGI 'lifespan' protocol appears unsupported.
INFO:     Application startup complete.
PASSED                [100%]INFO:     Waiting for application startup.
ERROR:    Exception in 'lifespan' protocol
Traceback (most recent call last):
  File "/home/lotso/PycharmProjects/uvicorn/uvicorn/lifespan/on.py", line 55, in main
    await app(scope, self.receive, self.send)
  File "/home/lotso/PycharmProjects/uvicorn/uvicorn/middleware/proxy_headers.py", line 45, in __call__
    return await self.app(scope, receive, send)
  File "/home/lotso/PycharmProjects/uvicorn/tests/test_lifespan.py", line 161, in asgi3app
    assert scope == {"version": "3.0", "spec_version": "2.0"}
AssertionError: assert {'asgi': {'spec_version': '2.0', 'version': '3.0'}, 'type': 'lifespan'} == {'spec_version': '2.0', 'version': '3.0'}
ERROR:    Application startup failed. Exiting.
```

### Expected behavior

test should fail

### Actual behavior

<!-- A clear and concise description of what actually happens. -->

### Debugging material

<!-- Any tracebacks, screenshots, etc. that can help understanding the problem.

NOTE:
- Please list tracebacks in full (don't truncate them).
- If relevant, consider turning on DEBUG or TRACE logs for additional details (see the Logging section on https://www.uvicorn.org/settings/ specifically the `log-level` flag).
- Consider using `<details>` to make tracebacks/logs collapsible if they're very large (see https://gist.github.com/ericclemmons/b146fe5da72ca1f706b2ef72a20ac39d).
-->

### Environment

- OS / Python / Uvicorn version: just run `uvicorn --version`
- The exact command you're running uvicorn with, all flags you passed included. If you run it with gunicorn please do the same. If there is a reverse-proxy involved and you cannot reproduce without it please give the minimal config of it to reproduce.

### Additional context

<!-- Any additional information that can help understanding the problem.

Eg. linked issues, or a description of what you were trying to achieve. -->

--- a/tests/test_lifespan.py
+++ b/tests/test_lifespan.py
@@ -146,26 +146,33 @@ def test_lifespan_with_failed_startup(mode, raise_exception):
     loop.run_until_complete(test())
 
 
-@pytest.mark.parametrize("mode", ("auto", "on"))
-def test_lifespan_scope_asgi3app(mode):
+def test_lifespan_scope_asgi3app():
     async def asgi3app(scope, receive, send):
-        assert scope == {"version": "3.0", "spec_version": "2.0"}
+        assert scope == {
+            "type": "lifespan",
+            "asgi": {"version": "3.0", "spec_version": "2.0"},
+        }
 
     async def test():
-        config = Config(app=asgi3app, lifespan=mode)
+        config = Config(app=asgi3app, lifespan="on")
         lifespan = LifespanOn(config)
 
         await lifespan.startup()
+        assert not lifespan.startup_failed
+        assert not lifespan.error_occured
+        assert not lifespan.should_exit
         await lifespan.shutdown()
 
     loop = asyncio.new_event_loop()
     loop.run_until_complete(test())
 
 
-@pytest.mark.parametrize("mode", ("auto", "on"))
-def test_lifespan_scope_asgi2app(mode):
+def test_lifespan_scope_asgi2app():
     def asgi2app(scope):
-        assert scope == {"version": "2.0", "spec_version": "2.0"}
+        assert scope == {
+            "type": "lifespan",
+            "asgi": {"version": "2.0", "spec_version": "2.0"},
+        }
 
         async def asgi(receive, send):
             pass
@@ -173,7 +180,7 @@ def test_lifespan_scope_asgi2app(mode):
         return asgi
 
     async def test():
-        config = Config(app=asgi2app, lifespan=mode)
+        config = Config(app=asgi2app, lifespan="on")
         lifespan = LifespanOn(config)
 
         await lifespan.startup()

